addons/mail/static/src/models/composer/composer.js

```diff
@@ -270,6 +272,136 @@ function factory(dependencies) {
             await this.env.bus.trigger('do-action', { action, options });
         }
 
+        parseBody() {
+            const escapedAndCompactContent = escapeAndCompactTextContent(this.textInputContent);
+            let body = escapedAndCompactContent.replace(/&nbsp;/g, ' ').trim();
+            body = this._generateMentionsLinks(body);
+            body = parseAndTransform(body, addLink);
+            body = this._generateEmojisOnHtml(body);
+            return body;
+        }
+
+        // this function will send the message inside the chatbox
+        // and queue it for sending.
+        queueMessage() {
+            const message = {
+                id: this.env.models['mail.message'].getNextTemporaryId(),
+                attachment_ids: this.attachments.map(attachment => attachment.id),
+                body: this.parseBody(),
+                channel_ids: this.mentionedChannels.map(channel => channel.id),
+                message_type: 'comment',
+                partner_ids: this.recipients.map(partner => partner.id),
+                subject: (this.subjectContent) ? this.subjectContent : undefined,
+                author: [['link', this.env.messaging.currentPartner]],
+                originThread: [['insert', {
+                    id: this.thread.id,
+                    model: this.thread.model,
+                }]],
+            };
+            this.update({
+                queue: [['insert', message]]
+            });
+            this._reset();
+            if (!this.queueProcess) {
+                this.postQueueMessage();
+            }
+        }
+
+        async postQueueMessage() {
+            this.update({
+                queueProcess: true,
+            });
+            const message = this.queue[0];
+            await this.postMessageQueued(message);
+            this.update({
+                queue: [['unlink', message]]
+            });
+            if (this.queue.length) {
+                this.postQueueMessage();
+            } else {
+                this.update({
+                    queueProcess: false,
+                });
+            }
```

Comment:
this should always be reset in this method I think, the next call to post will reset it to true if necessary

Reply:
I m not sure about this. A queue should never start if there is already a process that is posting message.
Rapidly fire true/false on this guard could lead to start to process 2 queues and create duplicate messages. It's unlikely to happen but it could ?

Reply:
I think in this case it's usually better to do the opposite: always call "process" and it's the processing method that is checking itself whether it is already in progress or not. This way multiple calls are always safe. -> The safeguard is only in one place, the function to guard, instead of on every potential caller.

Reply:
This look a lot better and could make it easier to start/stop queue process (on error for eg)